File: odemapster-core-querytranslator/src/main/java/es/upm/fi/dia/oeg/obdi/core/querytranslator/QueryTranslationOptimizerFactory.java
package es.upm.fi.dia.oeg.obdi.core.querytranslator;


public class QueryTranslationOptimizerFactory extends
		QueryTranslationOptimizer {
	
	public static QueryTranslationOptimizer createQueryTranslationOptimizerC() {
		QueryTranslationOptimizer queryTranslationOptimizerC = new QueryTranslationOptimizer();
		queryTranslationOptimizerC.setSelfJoinElimination(false);
		queryTranslationOptimizerC.setUnionQueryReduction(true);
		queryTranslationOptimizerC.setSubQueryElimination(false);
		queryTranslationOptimizerC.setTransJoinSubQueryElimination(false);
		queryTranslationOptimizerC.setTransSTGSubQueryElimination(false);
		return queryTranslationOptimizerC;
	}

	public static QueryTranslationOptimizer createQueryTranslationOptimizerE() {
		QueryTranslationOptimizer queryTranslationOptimizerE = new QueryTranslationOptimizer();
		queryTranslationOptimizerE.setSelfJoinElimination(false);
		queryTranslationOptimizerE.setUnionQueryReduction(true);
		queryTranslationOptimizerE.setSubQueryElimination(true);
		queryTranslationOptimizerE.setTransJoinSubQueryElimination(true);
		queryTranslationOptimizerE.setTransSTGSubQueryElimination(true);
		return queryTranslationOptimizerE;
	}
	
	public static QueryTranslationOptimizer createQueryTranslationOptimizerFE() {
		QueryTranslationOptimizer queryTranslationOptimizerFE = new QueryTranslationOptimizer();
		queryTranslationOptimizerFE.setSelfJoinElimination(true);
		queryTranslationOptimizerFE.setUnionQueryReduction(true);
		queryTranslationOptimizerFE.setSubQueryElimination(true);
		queryTranslationOptimizerFE.setSubQueryAsView(false);
		queryTranslationOptimizerFE.setTransJoinSubQueryElimination(true);
		queryTranslationOptimizerFE.setTransSTGSubQueryElimination(true);
		return queryTranslationOptimizerFE;
	}
	
	public static QueryTranslationOptimizer createQueryTranslationOptimizerFC() {
		QueryTranslationOptimizer queryTranslationOptimizerFC = new QueryTranslationOptimizer();
		queryTranslationOptimizerFC.setSelfJoinElimination(true);
		queryTranslationOptimizerFC.setUnionQueryReduction(true);
		queryTranslationOptimizerFC.setSubQueryElimination(false);
		queryTranslationOptimizerFC.setTransJoinSubQueryElimination(false);
		queryTranslationOptimizerFC.setTransSTGSubQueryElimination(false);
		return queryTranslationOptimizerFC;
	}	
}

File: morph-base/src/main/scala/es/upm/fi/dia/oeg/morph/base/engine/QueryTranslationOptimizerFactory.scala
package es.upm.fi.dia.oeg.morph.base.engine

class QueryTranslationOptimizerFactory {

}

object QueryTranslationOptimizerFactory {
	def createQueryTranslationOptimizerC() : QueryTranslationOptimizer = {
		val queryTranslationOptimizerC = new QueryTranslationOptimizer();
		queryTranslationOptimizerC.selfJoinElimination = false;
		queryTranslationOptimizerC.unionQueryReduction = true;
		queryTranslationOptimizerC.subQueryElimination = false;
		queryTranslationOptimizerC.transJoinSubQueryElimination = false;
		queryTranslationOptimizerC.transSTGSubQueryElimination = false;
		queryTranslationOptimizerC;
	}

	def createQueryTranslationOptimizerE() : QueryTranslationOptimizer = {
		val  queryTranslationOptimizerE = new QueryTranslationOptimizer();
		queryTranslationOptimizerE.selfJoinElimination = false;
		queryTranslationOptimizerE.unionQueryReduction = true;
		queryTranslationOptimizerE.subQueryElimination = true;
		queryTranslationOptimizerE.transJoinSubQueryElimination=true;
		queryTranslationOptimizerE.transSTGSubQueryElimination=true;
		queryTranslationOptimizerE;
	}
	
	def createQueryTranslationOptimizerFE() : QueryTranslationOptimizer  = {
		val queryTranslationOptimizerFE = new QueryTranslationOptimizer();
		queryTranslationOptimizerFE.selfJoinElimination = true;
		queryTranslationOptimizerFE.unionQueryReduction = true;
		queryTranslationOptimizerFE.subQueryElimination = true;
		queryTranslationOptimizerFE.subQueryAsView = false;
		queryTranslationOptimizerFE.transJoinSubQueryElimination = true;
		queryTranslationOptimizerFE.transSTGSubQueryElimination =true ;
		queryTranslationOptimizerFE;
	}
	
	def createQueryTranslationOptimizerFC() : QueryTranslationOptimizer = {
		val queryTranslationOptimizerFC = new QueryTranslationOptimizer();
		queryTranslationOptimizerFC.selfJoinElimination = true;
		queryTranslationOptimizerFC.unionQueryReduction = true;
		queryTranslationOptimizerFC.subQueryElimination = false;
		queryTranslationOptimizerFC.transJoinSubQueryElimination = false;
		queryTranslationOptimizerFC.transSTGSubQueryElimination = false;
		queryTranslationOptimizerFC;
	}  
}